File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int slow = 0;

        //First pass: move non-zero elements forward
        for(int fast = 0; fast < nums.size(); fast++){
            if(nums[fast] != 0){
                nums[slow] = nums[fast];
                slow++;
            }
        }

        //Second pass: fill the rest of the array with zeros
        for(int i = slow; i < nums.size(); i++){
            nums[i] = 0;
        }


        
    }
};

File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        length = len(nums)
        slow = 0

        for fast in range(0, length):
            if nums[fast] != 0:
                nums[slow] = nums[fast]
                slow += 1
        
        for i in range(slow, length):
            nums[i]=0


        